src/frontend/src/index.ts

```diff
@@ -94,6 +97,28 @@ const init = async () => {
   // Prepare the actor/connection to talk to the canister
   const connection = new Connection(readCanisterId());
 
+  // Figure out if user is trying to add a device. If so, use the anchor from the URL.
+  const addDeviceAnchor = getAddDeviceAnchor();
+  if (addDeviceAnchor !== undefined) {
+    // Register this device (tentatively)
+    await registerTentativeDevice(addDeviceAnchor, connection);
+
+    // Show a good bye message
+    await new Promise((_) => {
```

Comment:
not sure I get why the `new Promise` here, it's necessary? there is also no `resolve()` within, its ok?

Reply:
That's a very good question; copy pasta probably. Will remove